### ISSUE TITLE
Support PoA chains

### ISSUE THREAD

**Issue by vpulim** [ethereumjs/ethereumjs-monorepo#689]
In ``Blockchain.prototype._putBlockOrHeader``, it is assumed that the block is from a PoW chain:
```
    self.ethash.verifyPOW(block, function (valid) {
      next(valid ? null : new Error('invalid POW'))
    })
```

We need to properly handle PoA blocks. For now, we can simple skip the ``verifyPow`` call when ``block._common.consensus(hardfork) !== 'pow'`` 

Also, depends on https://github.com/ethereumjs/ethereumjs-block/issues/45

**Comment by holgerd77** [comment]
@jochem-brouwer following your linked issue from above, should we put this more prominently on the table (maybe not immediately but at least mid-term)? 🤔

Would be helpful if someone could collect some necessary base information and references to prepare for this (maybe directly by editing the initial comment for better visibility, pretty sure @vpulim will be ok with this).

**Comment by jochem-brouwer** [comment]
Linking [EIP-225: Clique proof-of-authority consensus protocol](https://eips.ethereum.org/EIPS/eip-225)

**Comment by holgerd77** [comment]
Solved by #1032 #1074 #1088 and #1103. 

Will close.
